src/studio/src/designer/backend.Tests/Services/RepositorySITests.cs

```diff
@@ -158,29 +158,37 @@ public async Task CopyRepository_TargetExistsLocally_InitialCloneMoved()
+            string origRemoteRepo = "apps-test";
+            string origRepo = "apps-test-2021";
+            string workingRemoteRepositoryName = TestDataHelper.GenerateTestRepoName(origRemoteRepo);
+            string targetRepositoryName = TestDataHelper.GenerateTestRepoName(origRepo);
+            var workingRemoteDirPath = string.Empty;
+            var createdTargetRepoPath = string.Empty;
             try
             {
-            PrepareRemoteTestData(org, sourceRepository);
-            TestDataHelper.CleanUpRemoteRepository("ttd", "apps-test-2021");
-            await TestDataHelper.CleanUpReplacedRepositories(org, targetRepository, developer);
+                workingRemoteDirPath = await TestDataHelper.CopyRemoteRepositoryForTest(org, origRemoteRepo, workingRemoteRepositoryName);
+                createdTargetRepoPath = await TestDataHelper.CopyRepositoryForTest(org, origRepo, developer, targetRepositoryName);
+                PrepareRemoteTestData(org, workingRemoteRepositoryName);
+                TestDataHelper.CleanUpRemoteRepository(org, targetRepositoryName);
+                await TestDataHelper.CleanUpReplacedRepositories(org, targetRepositoryName, developer);
 
-            RepositorySI sut = GetServiceForTest(developer);
+                RepositorySI sut = GetServiceForTest(developer);
 
-            // Act
-            await sut.CopyRepository(org, sourceRepository, targetRepository, developer);
+                // Act
+                await sut.CopyRepository(org, workingRemoteRepositoryName, targetRepositoryName, developer);
 
-            // Assert
-            string developerClonePath = Path.Combine(TestDataHelper.GetTestDataRepositoriesRootDirectory(), developer, org);
-            int actualCloneCount = Directory.GetDirectories(developerClonePath).Count(d => d.Contains("apps-test-2021"));
-            Assert.Equal(2, actualCloneCount);
+                // Assert
+                string developerClonePath = Path.Combine(TestDataHelper.GetTestDataRepositoriesRootDirectory(), developer, org);
+                int actualCloneCount = Directory.GetDirectories(developerClonePath).Count(d => d.Contains(targetRepositoryName));
+                Assert.Equal(2, actualCloneCount);
             }
             finally
             {
-                TestDataHelper.CleanUpRemoteRepository("ttd", "apps-test-2021");
-                await TestDataHelper.CleanUpReplacedRepositories(org, targetRepository, developer);
-                CleanUpRemoteTestData(org, sourceRepository);
+                TestDataHelper.CleanUpRemoteRepository(org, targetRepositoryName);
+                await TestDataHelper.CleanUpReplacedRepositories(org, targetRepositoryName, developer);
+                CleanUpRemoteTestData(org, workingRemoteRepositoryName);
+                TestDataHelper.DeleteDirectory(createdTargetRepoPath);
+                TestDataHelper.DeleteDirectory(workingRemoteDirPath);
```

Comment:
Any reason we dont use this method directly: `public static void Delete(string path, bool recursive)`?